mtec/integrationpluginmtec.cpp

```diff
@@ -0,0 +1,204 @@
+                    msg += item.value()->getHostAddress().toString();
+                    qCDebug(dcMTec()) << msg;
+                }
+            }
+
+            if (found == false) {
+                qCDebug(dcMTec()) << "Creating M_Tec object";
+
+                /* Create new MTec object and store it in hash table */
+                MTec *mtec = new MTec(hostAddress, this);
+                m_mtecConnections.insert(thing, mtec);
+
+                /* Store thing info in hash table */
+                m_mtecInfos.insert(thing, info);
+
+                msg = "Added M-Tec heatpump with address ";
+                msg += hostAddress.toString();
+                info->thing()->setStateValue(mtecConnectedStateTypeId, true);
+                info->finish(Thing::ThingErrorNoError, msg.toStdString().c_str());
+            } else {
+                info->finish(Thing::ThingErrorThingInUse, "IP address already in use");
+            }
+        } else {
+            info->finish(Thing::ThingErrorInvalidParameter, "No IP address given");
+        }
+    }
+}
+
+void IntegrationPluginMTec::postSetupThing(Thing *thing)
+{
+    qCDebug(dcMTec()) << "postSetupThing called";
+
+    if (!m_refreshTimer) {
+        m_refreshTimer = hardwareManager()->pluginTimerManager()->registerTimer(10);
+        connect(m_refreshTimer, &PluginTimer::timeout, this, &IntegrationPluginMTec::onRefreshTimer);
+    }
+
+    if (thing->thingClassId() == mtecThingClassId) {
+        qCDebug(dcMTec()) << "Thing id: " << thing->id();
+        MTec *mtec = m_mtecConnections.value(thing);
+
+        if (mtec != nullptr) {
+            connect(mtec, &MTec::statusUpdated, this, &IntegrationPluginMTec::onStatusUpdated);
+
+            qCDebug(dcMTec()) << "Thing set up, calling update";
+            update(thing);
+
+            thing->setStateValue(mtecConnectedStateTypeId, true);
+        }
+    }
+}
+
+void IntegrationPluginMTec::thingRemoved(Thing *thing)
+{
+    if (m_mtecConnections.contains(thing)) {
+        m_mtecConnections.take(thing)->deleteLater();
+    }
+}
+
+void IntegrationPluginMTec::executeAction(ThingActionInfo *info)
+{
+    Thing *thing = info->thing();
+    Action action = info->action();
+
+    if (thing->thingClassId() == mtecThingClassId) {
+      /* if (action.actionTypeId() == mtecPowerActionTypeId) { */
+          
+      /* } else { */
+      /*   Q_ASSERT_X(false, "executeAction", QString("Unhandled action: %1").arg(action.actionTypeId().toString()).toUtf8()); */
+      /*   } */
+    } else {
+        Q_ASSERT_X(false, "executeAction", QString("Unhandled thingClassId: %1").arg(thing->thingClassId().toString()).toUtf8());
+    }
+}
+
+void IntegrationPluginMTec::update(Thing *thing)
+{
+    if (thing->thingClassId() == mtecThingClassId) {
+        qCDebug(dcMTec()) << "Updating thing";
+
+        MTec *mtec = m_mtecConnections.value(thing);
+        Q_UNUSED(mtec);
+
+        if (mtec != nullptr) {
```

Comment:
For code consistancy:

    if (!mtec) 
        return;
    
    mtec->onRequestStatus();

Reply:
Done